### PR TITLE
feat(diffview): add diffview integrations

### PR DESCRIPTION
**Pull request by m00p1ng**
Added integration for [diffview.nvim](https://github.com/sindrets/diffview.nvim)

Before
Screenshot 2024-04-30 at 16 50 45

After
Screenshot 2024-04-30 at 16 53 10

### CODE REVIEW

**Comment by mrtnvgr** [comment]
I can't figure out what was changed based on these screenshots, they look identical for me 😕

**Comment by vollowx** [comment]
> I can't figure out what was changed based on these screenshots, they look identical for me 😕

@mrtnvgr The `M` at the signcolumn of the sidebar has been yellow following the 'modified color', and the title at the sidebar has been mauve following the theme color, as I can see.

**Review by mrtnvgr — APPROVED** [review]
👍

File: lua/catppuccin/groups/integrations/diffview.lua [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		DiffviewNonText = { link = "NonText" },
+		DiffviewCursorLine = { link = "CursorLine" },
+		DiffviewWinSeparator = {
+			fg = O.transparent_background and C.surface1 or C.base,
+			bg = O.transparent_background and C.none or C.base,
+		},
+		DiffviewSignColumn = { link = "Normal" },
+		DiffviewStatusLine = { link = "StatusLine" },
+		DiffviewStatusLineNC = { link = "StatusLineNC" },
+		DiffviewEndOfBuffer = { link = "EndOfBuffer" },
```
Comment (vollowx) [inline lua/catppuccin/groups/integrations/diffview.lua:18]: As they are linked in diffview by default, maybe these code can be removed? There're also code like this below and above.

File: lua/catppuccin/groups/integrations/diffview.lua [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		DiffviewStatusTypeChange = { fg = C.yellow },
+		DiffviewStatusUnmerged = { fg = C.yellow },
+		DiffviewStatusUnknown = { fg = C.red },
+		DiffviewStatusDeleted = { fg = C.red },
+		DiffviewStatusBroken = { fg = C.red },
+		DiffviewStatusIgnored = { fg = C.overlay0 },
+		DiffviewDiffAdd = { link = "DiffAdd" },
+		DiffviewDiffDelete = { link = "DiffDelete" },
+		DiffviewDiffChange = { link = "DiffChange" },
+		DiffviewDiffText = { link = "DiffText" },
```
Comment (vollowx) [inline lua/catppuccin/groups/integrations/diffview.lua:47]: Check above.

File: lua/catppuccin/groups/integrations/diffview.lua [hunk 3]
```diff
@@ -0,0 +1,51 @@
+local M = {}
+
+function M.get()
+	return {
+		DiffviewDim1 = { link = "Comment" },
+		DiffviewPrimary = { fg = C.blue },
+		DiffviewSecondary = { fg = C.green },
+		DiffviewNormal = { fg = C.text, bg = O.transparent_background and C.none or C.mantle },
+		DiffviewNonText = { link = "NonText" },
+		DiffviewCursorLine = { link = "CursorLine" },
```
Comment (vollowx) [inline lua/catppuccin/groups/integrations/diffview.lua:10]: Check below.

**Review by vollowx — COMMENTED** [review]
Beside this LGTM ❤️

File: lua/catppuccin/groups/integrations/diffview.lua (hunk 1 above)
Comment (m00p1ng) [reply lua/catppuccin/groups/integrations/diffview.lua]: Thank you for your comment. I have removed the code that you mentioned above ❤️

**Pull request merged**